core/src/main/java/io/questdb/griffin/SqlOptimiser.java

```diff
@@ -2309,6 +2341,14 @@ private int findColumnIdxByAst(ObjList<ExpressionNode> groupByNodes, ExpressionN
         return -1;
     }
 
+    private boolean findIfnodeExpressionContainsChildTableRef(ExpressionNode node, QueryModel targetModel) throws SqlException {
+        ChildTableColumnFinder childTableColumnFinder = new ChildTableColumnFinder(targetModel, false);
```

Comment:
Please reuse `ChildTableColumnFinder` object between query compilations. To achieve that, it's enough to introduce a new field in `SqlOptimiser`.